experimental/ConsoleApp6/ConsoleApp6/Api.cs

```diff
@@ -0,0 +1,118 @@
+            Metric.Record(measurement, labelName1, labelValue1);
+        public void Add(int measurement, string labelName1, string labelValue1, string labelName2, string labelValue2) =>
+            Metric.Record(measurement, labelName1, labelValue1, labelName2, labelValue2);
+        public void Add(int measurement, string labelName1, string labelValue1, string labelName2, string labelValue2, string labelName3, string labelValue3) =>
+            Metric.Record(measurement, labelName1, labelValue1, labelName2, labelValue2, labelName3, labelValue3);
+        public void Add(int measurement, string labelName1, string labelValue1, string labelName2, string labelValue2, string labelName3, string labelValue3,
+            params string[] additionalLabels) =>
+            Metric.Record(measurement, labelName1, labelValue1, labelName2, labelValue2, labelName3, labelValue3, additionalLabels);
+        public void Add(int measurement, LabelSet labels) =>
+            Metric.Record(measurement, labels);
+    }
+
+    public struct Gauge
+    {
+        internal Gauge(Metric metric) { Metric = metric; }
+        private Metric Metric { get; }
+        public void Set(int measurement) =>
+            Metric.Record(measurement);
+        public void Set(int measurement, string labelName1, string labelValue1) =>
+            Metric.Record(measurement, labelName1, labelValue1);
+        public void Set(int measurement, string labelName1, string labelValue1, string labelName2, string labelValue2) =>
+            Metric.Record(measurement, labelName1, labelValue1, labelName2, labelValue2);
+        public void Set(int measurement, string labelName1, string labelValue1, string labelName2, string labelValue2, string labelName3, string labelValue3) =>
+            Metric.Record(measurement, labelName1, labelValue1, labelName2, labelValue2, labelName3, labelValue3);
+        public void Set(int measurement, string labelName1, string labelValue1, string labelName2, string labelValue2, string labelName3, string labelValue3,
+            params string[] additionalLabels) =>
+            Metric.Record(measurement, labelName1, labelValue1, labelName2, labelValue2, labelName3, labelValue3, additionalLabels);
+        public void Set(int measurement, LabelSet labels) =>
+            Metric.Record(measurement, labels);
+    }
+
+
+    //TODO: ReadOnlyDictionary, ImmutableDictionary, or something else?
+    public class LabelSet : ReadOnlyDictionary<string, string>
+    {
+        public LabelSet(IDictionary<string, string> labels) : base(labels) { }
+        public LabelSet(string labelName1, string labelValue1) : base(null) { }
+        public LabelSet(string labelName1, string labelValue1, string labelName2, string labelValue2) : base(null) { }
+        public LabelSet(string labelName1, string labelValue1, string labelName2, string labelValue2, string labelName3, string labelValue3) : base(null) { }
+    }
+
+
+
+    public static class MetricAggregations
```

Comment:
Wondering if any Aggregations should be part of the API.  If it is, how would new aggregations be added?

Reply:
These are just convenient singletons to reduce allocations. The intent is that anyone can derive new types and invoke new to create an instance of their new type.